Firebase/Auth/Source/User/FIRUser.m

```diff
@@ -791,6 +792,21 @@ - (void)reauthenticateAndRetrieveDataWithCredential:(FIRAuthCredential *) creden
   });
 }
 
+- (void)reauthenticateWithProvider:(id<FIRFederatedAuthProvider>)provider
+                        UIDelegate:(nullable id<FIRAuthUIDelegate>)UIDelegate
+                        completion:(nullable FIRAuthDataResultCallback)completion {
+#if TARGET_OS_IOS
```

Comment:
Since this only works on iOS, can we mark the method in the public header as unavailable?

I believe this should work: `API_UNAVAILABLE(macos, tvos)`

Reply:
Yeah, sounds good. I added `API_AVAILABLE(ios)`.